markdown/treeprocessors.py

```diff
@@ -429,3 +431,28 @@ def run(self, root):
                 # Only prettify code containing text only
                 if not len(code) and code.text is not None:
                     code.text = util.AtomicString(code.text.rstrip() + '\n')
+
+
+class UnescapeTreeprocessor(Treeprocessor):
+    """ Restore escaped chars """
+
+    RE = re.compile(r'{}(\d+){}'.format(util.STX, util.ETX))
+
+    def _unescape(self, m):
+        return chr(int(m.group(1)))
+
+    def unescape(self, text):
+        return self.RE.sub(self._unescape, text)
+
+    def run(self, root):
+        """ Loop over all elements and unescape all text. """
+        for elem in root.iter():
+            # Unescape text content
+            if elem.text and not elem.tag == 'code':
```

Comment:
I'm not sure we actually need to skip `code` tags, In fact, if I remove the check, the tests all pass. In fact, the previous code did not have a way to distinguish between code and other content. However, there is always a possibility that `code` could intentionally contain what looks like a placeholder. In that case, the content should not be altered. Therefore, I have left the check in.